pytorch_lightning/utilities/fetching.py

```diff
@@ -385,3 +398,9 @@ def fetching_function(self) -> Tuple[int, Iterator]:
         if not self.done:
             return self.fetched, self.iterator
         raise StopIteration
+
+    def on_fetch_start(self) -> None:
```

Comment:
These are only called in `DataFetcher._fetch_next_batch` but that method is not used by this class (this class subclasses `AbstractDataFetcher`)

Particularly, this start and stop needs to wrap the `next` in `StepFuncDataLoaderIter`